Analysis/bin/ShuffleMergeSpectral.cxx

```diff
@@ -0,0 +1,681 @@
+          const std::string& file_name = file_names.at(*current_file_index);
+          current_tuple.reset();
+          current_file = root_ext::OpenRootFile(file_name);
+          current_tuple = std::make_shared<TauTuple>("taus", current_file.get(), true, disabled_branches);
+          entries_file = current_tuple->GetEntries();
+          if(!entries_file)
+            throw analysis::exception("Root file inside %1% is empty.") % name;
+      }
+      ++total_n_processed;
+      current_tuple->GetEntry(current_n_processed++);
+      (*current_tuple)().dataset_id = std::hash<std::string>{}(name);
+      (*current_tuple)().dataset_group_id = std::hash<std::string>{}(group_name);
+      return true;
+    }
+
+    const Tau& GetNextTau()
+    {
+      return current_tuple->data();
+    }
+
+    size_t GetNumberOfEvents() const { return entries_file; }
+    const std::string& GetName() const { return name; }
+    const std::vector<std::string>& GetFileNames() const { return file_names; }
+    double GetWeight() const { return weight; }
+    // const std::string& GetBinName() const { return bin_name; }
+    // void SetBinName(const std::string& _bin_name) { bin_name = _bin_name; }
+
+  private:
+      const std::string name;
+      const std::string group_name;
+      const std::vector<std::string> file_names;
+      const std::set<std::string> disabled_branches;
+      const double weight;
+      std::shared_ptr<TFile> current_file;
+      std::shared_ptr<TauTuple> current_tuple;
+      boost::optional<ULong64_t> current_file_index;
+
+      // types should be checked
+      ULong64_t current_n_processed;
+      ULong64_t files_n_total;;
+      size_t entries_file;
+      size_t total_n_processed;
+  };
+
+
+struct EntryDesc {
+    using TauType = analysis::TauType;
+    using SampleType = analysis::SampleType;
+
+    std::string name;
+    std::map< std::string, std::vector<std::string>> data_files;
+    std::map< std::string, std::string> spectrum_files;
+    std::vector< std::string > tau_types;
+    double weight;
+
+    EntryDesc(const analysis::PropertyConfigReader::Item& item,
+              const std::string& base_input_dir,
+              const std::string& base_spectrum_dir)
+    {
+        using boost::regex;
+        using boost::regex_match;
+        using boost::filesystem::path;
+        using boost::make_iterator_range;
+        using boost::filesystem::directory_iterator;
+        using boost::filesystem::is_directory;
+        using boost::filesystem::is_regular_file;
+        using boost::split;
+
+        name = item.name;
+        std::cout << name << std::endl;
+        const std::string dir_pattern_str = item.Get<std::string>("dir");
+        const std::string file_pattern_str = item.Get<std::string>("file");
+        weight = item.Has("weight") ? item.Get<double>("weight") : 1;
+        const std::string tau_types_str = item.Get<std::string>("types");
+        split(tau_types, tau_types_str, boost::is_any_of(","));
```

Comment:
Better to use `analysis:: SplitValueList` or `analysis::SplitValueListT`, which is a commodity wrapper around `boost::split`
https://github.com/cms-tau-pog/TauMLTools/blob/master/Core/interface/TextIO.h#L112-L125
```c++
tau_types = SplitValueListT<TauType, std::set<TauType>>(tau_types_str, false, ",");
```